### PR TITLE
[automatic failover] Builder APIs for DatabaseConfig and CircuitBreakerConfig  (CAE-1695)

### PR DESCRIPTION
**Pull request by ggivo**
## Overview

This PR introduces fluent Builder APIs for `DatabaseConfig` and `CircuitBreakerConfig` to improve the developer experience when configuring multi-database clients with automatic failover capabilities.

### 1. **DatabaseConfig Builder API** (#3571)
- Added `DatabaseConfig.Builder` with fluent API for configuring database endpoints
- Builder methods:
  - `weight(float)` - Set load balancing weight (default: 1.0)
  - `circuitBreakerConfig(CircuitBreakerConfig)` - Set circuit breaker configuration
  - `healthCheckStrategySupplier(HealthCheckStrategySupplier)` - Set health check strategy (default: `PingStrategy.DEFAULT`)
- Removed public constructors in favor of builder pattern
- Changed default health check from `null` to `PingStrategy.DEFAULT`
- Introduced `HealthCheckStrategySupplier.NO_HEALTH_CHECK` constant for explicitly disabling health checks
- **Removed `clientOptions` field** from `DatabaseConfig` as it was not being applied (tracked in #3572)

### 2. **CircuitBreakerConfig Builder API**
- Added `CircuitBreakerConfig.Builder` with fluent API for circuit breaker configuration
- Builder methods:
  - `failureThreshold(int)` - Set failure threshold
  - `successThreshold(int)` - Set success threshold  
  - `windowSize(Duration)` - Set sliding window size (minimum: 2 seconds)
  - `halfOpenMaxAttempts(int)` - Set max attempts in half-open state
  - `openStateDuration(Duration)` - Set duration to stay in open state
  - `addTrackedExceptions(Class<?>...)` - Add exception types to track
  - `trackedExceptions(Set<Class<?>>)` - Set all tracked exceptions
- Enforced minimum window size of 2 seconds with validation
- Tracked exceptions are now non-null (empty set by default)

### 3. **Test Coverage**
- Added comprehensive unit tests:
  - `DatabaseConfigBuilderUnitTests` - 11 tests covering all builder functionality
  - `CircuitBreakerConfigBuilderUnitTests` - 15 tests covering all builder functionality and edge cases
- Updated integration tests to use new builder APIs

### DatabaseConfig Builder
```java
        DatabaseConfig config1 = DatabaseConfig.builder(RedisURI.create("redis://localhost:6379"))
                .weight(1.0f)
                .circuitBreakerConfig(CircuitBreaker.CircuitBreakerConfig.builder()
                        .failureRateThreshold(10.0f)
                        .minimumNumberOfFailures(5)
                        .metricsWindowSize(5)
                        .build())
                .healthCheckStrategySupplier(PingStrategy.DEFAULT)
                .build();

        // Use default circuit breaker config and default PING health check
        DatabaseConfig config2 = DatabaseConfig.builder(RedisURI.create("redis://localhost:6379"))
                .weight(0.5f)
                .build();

        MultiDbClient multiDbclient = MultiDbClient.create(Arrays.asList(config1, config2));
```

### CODE REVIEW

File: src/main/java/io/lettuce/core/failover/CircuitBreaker.java [hunk 1]
```diff
@@ -143,6 +154,131 @@ public String toString() {
...
+             * @param exceptionClasses one or more exception classes to track, must not be {@code null} and must not contain
+             *        {@code null} elements
+             * @return {@code this} builder
+             * @since 7.4
+             */
+            @SafeVarargs
+            public final Builder addTrackedExceptions(Class<? extends Throwable>... exceptionClasses) {
+                LettuceAssert.notNull(exceptionClasses, "Exception classes must not be null");
+                LettuceAssert.noNullElements(exceptionClasses, "Exception classes must not contain null elements");
+                LettuceAssert.notEmpty(exceptionClasses, "Exception classes must contain at least one element");
```
Comment (atakavci) [inline src/main/java/io/lettuce/core/failover/CircuitBreaker.java:225]: i dont think we need assertions other than `notNull(exceptionClasses...` but if we'd like to have them, then why not put those into `trackedExceptions(Set..` as well.

File: src/main/java/io/lettuce/core/failover/DatabaseConfig.java [hunk 2]
```diff
@@ -1,83 +1,51 @@
...
-     * @param weight the weight for load balancing, must be greater than 0
+     * @param builder the builder
      */
-    public DatabaseConfig(RedisURI redisURI, float weight) {
-        this(redisURI, weight, null, null, null);
+    DatabaseConfig(Builder builder) {
+        this.redisURI = builder.redisURI;
+        this.weight = builder.weight;
+        this.circuitBreakerConfig = builder.circuitBreakerConfig != null ? builder.circuitBreakerConfig
+                : CircuitBreakerConfig.DEFAULT;
```
Comment (atakavci) [inline src/main/java/io/lettuce/core/failover/DatabaseConfig.java:47]: nit: this feels like it belongs to builder.

**Comment by ggivo** [comment]
merged the PR accidentally before pushing the changes to address comments. Will open new PR to address them shortly
